.github/workflows/run-linting-tests.yml

```diff
@@ -0,0 +1,42 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Linting tests
+
+on:
+  push:
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    name: Linting Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e '.[dev]'
+      - name: Check formatting
+        run: |
+          isort -c .
+          black --check .
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 .
+      #
+      # disabled mypy for now, FIXME
+      #
+      # - name: Check with mypy
+      #  run: |
+      #    mypy src/supa
```

Comment:
What are the errors? It should be relatively simple to fix them, with the correct type: ignore statements, or they really do indicate a bug.

Reply:
Judging by the last failed run, you should be able to simply exclude main in the run, and the other errors seem to indicate valid potential bugs.

Reply:
Good point. Fixing the typing errors wasn't really hard, even without ignoring stuff. Enabled mypy again.